### PR TITLE
Add support for JSON API sparse fieldsets #9132

### PR DESCRIPTION
**Pull request by caudatecoder**
https://github.com/spree/spree/issues/9132

### CODE REVIEW

File: api/spec/controllers/spree/api/v2/base_controller_spec.rb [hunk 1]
```diff
@@ -46,4 +46,36 @@ def default_resource_includes
...
+      end
+
+      it 'returns nil' do
+        expect(dummy_controller.send(:sparse_fields)).to eq(nil)
+      end
+    end
+
+    context 'with invalid params format' do
+      before do
+        dummy_controller.params = { fields: 'variants,images,taxons' }
```
Comment (damianlegawiec) [inline api/spec/controllers/spree/api/v2/base_controller_spec.rb]: @caudatecoder shouldn't this be stuff like `slug`, `name` etc?

Comment (caudatecoder) [reply api/spec/controllers/spree/api/v2/base_controller_spec.rb]: Yeah it will be closer to the real payload

**Comment by VygovskySergey** [comment]
@caudatecoder What do you think about moving this method to a separate service and use as dependency?

**Comment by caudatecoder** [comment]
@VygovskySergey IMHO a method in the `BaseController` is easier to reuse and since it's a part of `JSON:API` specification it should be as easy to use as possible

**Review by damianlegawiec — APPROVED** [review]

**Comment by VygovskySergey** [comment]
@caudatecoder Okay, make sense

**Review by VygovskySergey — APPROVED** [review]

**Pull request merged**